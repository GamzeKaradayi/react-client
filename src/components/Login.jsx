import React  from 'react';
import PropTypes from 'prop-types';

 const  Login = (props) => {
	const test=()=>{
		console.log(this.refs.name);
	}
	return(
	<div className="login-page">
		<div className="title">
			<div className="bold">öğrenci</div>
			<div>evi</div>
		</div>
		<div className="sub-title">
			ev idaresi uygulaması
		</div>
		<div className="login-form">
			<input className="common-input" type="text" value={props.name} placeholder="Ad" name='name' onChange={(e)=>{props.handleChange("name",e.target.value)}} />
			<input className="common-input" type="text" value={props.surname} placeholder="Soyad" name='surname' onChange={(e)=>{props.handleChange("surname",e.target.value)}} />
			<select onChange={(e)=>{props.handleChange("selectedHome",e.target.value)}} className="common-input">{!props.selectedHome.id ?
			<option selected disabled>Ev Seçiniz</option>
			: null}
				{props.homes.map((home)=>{
					return <option selected={props.selectedHome.id===home.id} value={home.id}>{home.name}</option>
				})}
			</select>
			<div className="login-button-container">
				<button onClick={test} className="login-button">GİRİŞ</button>
			</div>
		</div>
	</div>);
}
Login.propTypes={
	onClick:PropTypes.func,
	homes:PropTypes.array
}
export default Login;
