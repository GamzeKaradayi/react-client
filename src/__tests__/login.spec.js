import React from 'react';
import {shallow} from 'enzyme';
import Login from '../components/Login';

const houses=[
    {
        "id": 1,
        "name": "houseA"
    },
    {
        "id": 2,
        "name": "houseB"
    },
    {
        "id": 3,
        "name": "houseC"
    },
    {
        "id": 4,
        "name": "houseD"
    }
]

test('Login component renders correctly with correct props', () => {

	const login = shallow(
	<Login login={()=>{}}
		homes={houses}
		handleChange={()=>{}}
		name={"testUserName"}
		surName={"testLastName"}
		selectedHome={houses[0]}
		loadingState={false}
		validate={true}
	/>);

	expect(login.find('input').length).toEqual(2);
	expect(login.find('button').length).toEqual(1);
	expect(login.find('option').length).toEqual(4);
	expect(login.find('help-block').length).toEqual(0);
  });

  test('Login component renders correctly with missing props', () => {

	const login = shallow(
	<Login login={()=>{}}
		homes={houses}
		handleChange={()=>{}}
		name={null}
		surName={null}
		selectedHome={houses[0]}
		loadingState={false}
		validate={true}
	/>);

	expect(login.find('input').length).toEqual(2);
	expect(login.find('button').length).toEqual(1);
	expect(login.find('option').length).toEqual(4);
	expect(login.find('.help-block').length).toEqual(2);

  });
