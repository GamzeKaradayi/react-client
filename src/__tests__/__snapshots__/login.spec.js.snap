// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component matches snapshot 1`] = `
<div
  className="login-page"
>
  <div
    className="title"
  >
    <div
      className="bold"
    >
      öğrenci
    </div>
    <div>
      evi
    </div>
  </div>
  <div
    className="sub-title"
  >
    ev idaresi uygulaması
  </div>
  <div
    className="login-form"
  >
    <div
      className="loader-container"
    />
    <input
      className="common-input"
      name="name"
      onChange={[Function]}
      placeholder="Ad"
      type="text"
      value="testUserName"
    />
    <input
      className="common-input"
      name="surName"
      onChange={[Function]}
      placeholder="Soyad"
      type="text"
      value={undefined}
    />
    <select
      className="common-input"
      onChange={[Function]}
    >
      <option
        selected={true}
        value={1}
      >
        houseA
      </option>
      <option
        selected={false}
        value={2}
      >
        houseB
      </option>
      <option
        selected={false}
        value={3}
      >
        houseC
      </option>
      <option
        selected={false}
        value={4}
      >
        houseD
      </option>
    </select>
    <div
      className="login-button-container"
    >
      <button
        className="login-button"
        onClick={[Function]}
      >
        GİRİŞ
      </button>
    </div>
  </div>
</div>
`;

exports[`matches snapshot 1`] = `
<div
  className="login-page"
>
  <div
    className="title"
  >
    <div
      className="bold"
    >
      öğrenci
    </div>
    <div>
      evi
    </div>
  </div>
  <div
    className="sub-title"
  >
    ev idaresi uygulaması
  </div>
  <div
    className="login-form"
  >
    <div
      className="loader-container"
    />
    <input
      className="common-input"
      name="name"
      onChange={[Function]}
      placeholder="Ad"
      type="text"
      value="testUserName"
    />
    <input
      className="common-input"
      name="surName"
      onChange={[Function]}
      placeholder="Soyad"
      type="text"
      value={undefined}
    />
    <select
      className="common-input"
      onChange={[Function]}
    >
      <option
        selected={true}
        value={1}
      >
        houseA
      </option>
      <option
        selected={false}
        value={2}
      >
        houseB
      </option>
      <option
        selected={false}
        value={3}
      >
        houseC
      </option>
      <option
        selected={false}
        value={4}
      >
        houseD
      </option>
    </select>
    <div
      className="login-button-container"
    >
      <button
        className="login-button"
        onClick={[Function]}
      >
        GİRİŞ
      </button>
    </div>
  </div>
</div>
`;
